File: 6. Arbol/Arbol.c
#include <stdio.h> // imprimir
#include <math.h> // inicializar pseudoaleatorio
#include <stdlib.h> // malloc & free


#define MAXIMO 100

typedef struct nodoDeArbol {
  int valor;
  int numeroDeHijos;
  struct nodoDeArbol* padre;
  struct nodoDeArbol** hijos;
} nodo;


typedef struct arbolEnlazado {
    struct nodoDeArbol* raiz;  
}arbol;

int main(int argv, char** args){
    
  arbol* a = (arbol*)malloc(sizeof(arbol));
  a->raiz = NULL;
  nodo** aux = (nodo**)malloc(sizeof(arbol)*8);
  nodo* n = NULL;

  n = (nodo*)malloc(sizeof(nodo));

  e->valor = 0;
  e->padre = NULL;

  e->hijos = (nodo**)malloc(sizeof(2*nodo));
  e->hijos0[] = 3;
  e->hijos0[] = 3;
  e->numeroDeHijos = 2;
  
  
  return 0;
}

File: 6. Arbol/Arbol.py
class nodo:
    def __init__(self,valor):
        self.valor = valor;
        self.padre = None
        self.hijos =list()


    def recorre(self, prof):
        prefijo = ' ' * prof
        print ' %s %d ' & (prefijo, self.valor)
        for hijo in self.hijos:
            hijo.recorre(prof + 1)
        return

        
    def __str__(self):
        nombres = ''
        for hijo in self.hijos:
            nombres += '%d' % hijo.valor
        return '%d: %s' % (self.valor, nombre)

    def __repr__(self):
        return str(self)
        
    def agrega(self,valor,padre):
        if self.valor == padre:
            hijo = nodo(valor)
            self.hijos.append(hijo)
            hijo.padre = self
            return True
        else:
            for hijo in self.hijos:
                if hijo.agrega(valor,padre):
                    return True
            return False


raiz = nodo(3)

raiz.agrega(5,3)

raiz.recorre(0)
